src/sage/rings/semirings/tropical_mpolynomial.py

```diff
@@ -0,0 +1,374 @@
+            NotImplementedError: can only plot the graph of tropical 
+            multivariate polynomial in two variables
+        """
+        from sage.arith.srange import srange
+
+        if len(self.parent().variable_names()) != 2:
+            raise NotImplementedError("can only plot the graph of tropical" \
+                                " multivariate polynomial in two variables")
+        axes = self.tropical_variety()._axes()
+        xmin, xmax = axes[0][0], axes[0][1]
+        ymin, ymax = axes[1][0], axes[1][1]
+        step = 0.5
+        x_point = srange(xmin, xmax+step, step)
+        y_point = srange(ymin, ymax+step, step)
+        res = []
+        T = self.parent().base()
+        for x in x_point:
+            for y in y_point:
+                val = self(T(x),T(y)).lift()
+                res.append([x,y,val])
+        return list_plot3d(res, point_list=True)    
+
+    def tropical_variety(self):
+        r"""
+        Return tropical roots of ``self``.
+        
+        In multivariate case, the roots can be represented by a tropical
+        variety. For 2 dimensions, it is also called a tropical curve.
+
+        OUTPUT: ``TropicalVariety`` object
+        
+        EXAMPLES:
+
+        Tropical curve for tropical polynomials in two variables::
+
+            sage: T = TropicalSemiring(QQ, use_min=False)
+            sage: R.<x,y> = PolynomialRing(T)
+            sage: p1 = x + y + R(0); p1
+            0*x + 0*y + 0
+            sage: p1.tropical_variety()
+            Tropical curve of 0*x + 0*y + 0 are 
+            [[(t1, t1), [t1 >= 0], 1]
+            [(0, t1), [t1 <= 0], 1]
+            [(t1, 0), [t1 <= 0], 1]]
+
+        Finding tropical hypersurface for tropical polynomials in more than
+        two variables::
+
+            sage: T = TropicalSemiring(QQ)
+            sage: R.<x,y,z> = PolynomialRing(T)
+            sage: p1 = R(1)*x*y + R(-1/2)*x*z + R(4)*z^2; p1
+            1*x*y + (-1/2)*x*z + 4*z^2
+            sage: p1.tropical_variety()
+            Tropical surface of 1*x*y + (-1/2)*x*z + 4*z^2 are 
+            [[(t1, t2 - 3/2, t2), [t1 <= t2 + 9/2], 1]
+            [(2*t1 - t2 + 3, t2, t1), [t2 + 3/2 <= t1], 1]
+            [(t1 + 9/2, t2, t1), [t1 <= t2 + 3/2], 1]]
+        """
+        if self.parent().ngens() == 2:
+            return TropicalCurve(self)
+        if self.parent().ngens() == 3:
+            return TropicalSurface(self)
+        return TropicalVariety(self)
+
+class TropicalMPolynomialSemiring(UniqueRepresentation, Parent):
+    """
+    Semiring structure of tropical polynomials in multiple variables.
+    """
+
+    def __init__(self, base_semiring, n, names, order='degrevlex'):
+        """
+        EXAMPLES::
+
+            sage: T = TropicalSemiring(QQ, use_min=True)
+            sage: R.<x,y,z> = PolynomialRing(T)
+            sage: R(-1)*x + R(1)*x + y
+            (-1)*x + 0*y
+            sage: (x+y+z)^2
+            0*x^2 + 0*x*y + 0*y^2 + 0*x*z + 0*y*z + 0*z^2
```

Comment:
```suggestion
            sage: R.<x,y,z> = PolynomialRing(T)
            sage: TestSuite(R)
```
The other examples should be in the class-level docstring.